### ISSUE TITLE
Skip absence-data on indexing

### ISSUE THREAD

**Issue by kbraak** [gbif/portal16#308]
**_Relevant content transferred from http://dev.gbif.org/issues/browse/POR-2864, which was created 02/Oct/15_** 

### Description of problem

"Our current index is not able to interpret absence-data. For the time being, a published record to us means occurrence, ergo presence, not absence" - [AH](http://dev.gbif.org/issues/browse/DM-178?focusedCommentId=16070&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16070)

### How are users of GBIF.org affected?

"At the 8th EU Nodes workshop/meeting in Libson 18-21 April 2016 several participants reported that they noticed absence data been shown on GBIF.org." - [KB](http://dev.gbif.org/issues/browse/POR-2864?focusedCommentId=21960&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21960)

### How are data publishers affected? 

In the context of publishing sample event data, I instructed them to advise publishers in their network not to publish absence data through GBIF.org until we can handle it properly" - [KB](http://dev.gbif.org/issues/browse/POR-2864?focusedCommentId=21960&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21960)

### What does absence-data look like?

Publishers may be publishing occurrence records with absence-data in the following fields/ways:

    occurrenceStatus=absent
    individualCount=0
    organismQuantity=0

Instructions given to publishers on how to publish absences can be found in this [IPT FAQ](https://github.com/gbif/ipt/wiki/samplingEventData#q-how-do-i-publish-absence-data). 

### Hurdles and ideas for implementation

* "we should probably already avoid creating a verbatim record? Or even not a fragment? Or interpretation removes the verbatim and/or fragment?" Alternatively we could keep all records incl absent ones, interpret the OccurrenceStatus from the different options listed above and then avoid them in cubes & maps and offer an OccurrenceStatus filter in solr. Maybe the cube should even just have another dimension present/absent.  - [MD](http://dev.gbif.org/issues/browse/POR-2864?focusedCommentId=21384&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21384)
* "I would strongly prefer the second option, for 4 reasons: 
  1. Those records are useful (e.g. for modelling). 
  2. This is a representation issue. One could have them in the occurrence store, but remove them from maps (like how none georeferenced records are not shown), or as you suggest: create a filter/dimension for them and maybe shown them on maps in a different colour. 
  3. If absent records are removed during interpretation, it becomes much more difficult for the publisher to assess why the number of records in their source is different from the number of records presented on GBIF. That difference should indicate an issue in the process (e.g. records removed at the source are not deleted by GBIF, ...), not because valid absent records were ignored. 
  4. Absent records are going to become more ubiquitous, especially if GBIF can handle them nicely." - [PD](http://dev.gbif.org/issues/browse/POR-2864?focusedCommentId=21437&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21437)
* "As for counts shown in the portal I think we should:
  * show all counts on http://www.gbif.org/
  * show all & georeferenced presence count on http://www.gbif.org/occurrence/
  * use presence only counts for all the metrics below the main search block on http://www.gbif.org/occurrence/
  * show all counts in dataset details
  * show detailed breakdwon by presence & absence on dataset stats page
  * include all records in a default search, but allow an occurrenceStatus enum based filter
  * occurrenceStatus enum currently has 7 values which we might want to reduce to just present, absent, doubtful: https://github.com/gbif/gbif-api/blob/master/src/main/java/org/gbif/api/vocabulary/OccurrenceStatus.java#L23" - [MD](http://dev.gbif.org/issues/browse/POR-2864?focusedCommentId=21457&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21457)
* "Another issue is what do people do with the data when it's downloaded? Even if, say, GBIF hides absences on the maps, if that data is in the download then we are assuming that users are sufficiently savvy to handle absences correctly. if not, this could have dire consequences for naive efforts to model distributions (e.g., seabirds in the Alps)." - [RP](http://dev.gbif.org/issues/browse/POR-2864?focusedCommentId=21478&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21478)
* "For downloads and similarly for API search/download calls for example via the R package we should simply default to exclude absence data, i.e. set an occurrenceStatus filter by default to include only presence. This way someone will only see absence data when they actively request it by changing the filter." - [MD](http://dev.gbif.org/issues/browse/POR-2864?focusedCommentId=21479&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21479)

**Comment by dhobern** [comment]
My view is that we need to be really careful about this issue.  Fields may end up containing "0" as the value for many reasons, not all of them because of absences - it may just represent a missing value in a database.

That means that we should probably flag occurrence records with an issue if they have 0 in these positions, but not treat them for now at least as absences. Maybe we should carry out a sampling of records with 0 in these fields and see whether they seem to represent presences or absences.

For sampling-event data, it is messier.  We should place the emphasis on the fact that a sample event without a record for a species recorded elsewhere in the same dataset will be considered to mean that the species was not found in the event.  We could/should have ways to represent sampling event datasets as sparse matrices and perhaps allow zeroes to be populated in these (as an option) during download.

I think that we should not encourage people to share special absence records except in cases where it is hard for them to clear these out from sampling event datasets.  They should not be part of our world in other cases.

**Comment by Tasilee** [comment]
This is my attempt at raising the issue of 'absences' (after feedback from Dmitry, Ben and 

Presence-Absence Data
Lee Belbin 8/2/2017
To an ecologist, ‘presence-absence data’ implies an absence of a species at a location. Presence-only data identifies that a species was present at a location. The most important misconception about presence-absence data is that absences are explicitly recorded in the field. This is rare, but the ALA does have some. Absences are usually inferred in most systematic surveys. 

A transect of pitfall traps for invertebrates or a suite of quarter hectare plots for trees are typical examples of systematic surveys. Pitfall traps are placed in the ground at regular intervals along a transect, left for a period and then species (often with numbers) that have fallen into the traps are recorded. With the tree survey, the presence or number of all tree species with a diameter at breast height above a threshold may be recorded. The key aspect of such surveys is specific or implied taxonomic scope – what type of species were in the scope of the survey.

The data from systematic surveys will usually take the form of “Site X, latitude, longitude, species A, species B, species C…”. Each site will also have metadata and possibly site environmental data. There is no mention of species that were not seen at the site. 

Absences can however be inferred by combining sites with similar survey type and taxonomic scope. The Darwin Core terms that may help to evaluate if sites can be meaningfully combined would include scientificName, class, order, family, vernacularName, samplingProtocol, establishmentMeans, occurrenceStatus, eventID, parentEventID, recordedBy, basisOfRecord, eventDate, sampleSizeUnit, samplingEffort, fieldNotes and eventRemarks (see http://tdwg.github.io/dwc/terms/).

Once candidate sites are identified, they can be combined by generating a sites by species matrix (=co-occurrence matrix). If we add two more sites to our Site X above, say “Site Y, species B, species D, species E” and “Site Z, species A, species D, species F”, we can generate the following table (matrix)  where “1” represents a presence and “0” represents an absence-

                 Species A    Species B    Species C    Species D    Species E    Species F
Site X             1                 1                 1                 0                  0                  0
Site Y             0                 1                 0                 1                  1                  1
Site Z             1                 0                 0                 1                  0                  1

The absences are now explicit, for example “Species D is absent from X”.  The great difficulty is however determining what sites can be meaningfully combined. Could survey sites for trees be combined if they were of different sized plots? Probably. Could pitfall survey sites be combined if they span 100 years? Depends. Surveys may use the same gear but target different species types or resolve taxa to different levels. Careful evaluation of available Darwin Core terms is needed before combining sites. 

Recommendations
1.	The use of Darwin Core Event terms must be strongly encouraged. The ALA has systematic survey data but records have not been fully documented to enable effective re-use. Some systematic survey records have been imported without samplingProtocol etc.
2.	samplingProtocol is a verbatim term. Values need to be classified into a useful subset. TDWG has initiated work on controlled vocabularies (Paula Zermoglio). 
3.	samplingProtocol and other Darwin Core terms are not currently indexed in the ALA, so sites can only be combined if they are exported with the full Darwin Core terms, evaluated and then processed. But what sites to export? Catch-22.
4.	A standard suite of core tests/assertions and all Darwin Core terms need to be indexed for filtering to be possible. GBIF and TDWG are developing a standard test suite (lead by me).
5.	Searching and filtering logic needs to support AND and OR. For example, “display sites that have (samplingProtocol=”pitfall” OR samplingProtocol=”most net”) and eventDate >=”1963-03-08T14:07-0600"1990 AND eventDate<"1999-03-08T14:07-0600".
6.	Create a new tool in the ALA’s Spatial Portal to generate a sites by species matrix from a set of filtered sites. This matrix can then be used in tools such as SDM where 'true absences' (rather than pseudo absences) can be utilized.

**Comment by albenson-usgs** [comment]
The current method of having the data provider filter out absence records in the IPT is untenable. This prevents other data aggregators that harvest from IPT (OBIS for instance) from being able to bring in the absence records and creates a disconnect between any measurement or facts and the occurrence IDs that are link to them. As an example, I just recently published a dataset that I aligned with Event Core and used Extended Measurement or Facts to make accessible information about coral and macroalgae percent cover in the Virgin Islands (https://www1.usgs.gov/obis-usa/ipt/manage/resource.do?r=noaa_coralreefmonitoring_lpipercentcover) if I filter out records in the occurrence file that have occurrenceStatus = absent then measurements in the extended measurements or facts file will not have a corresponding occurrence ID to link to.
![mof](https://cloud.githubusercontent.com/assets/7586150/24302669/00e937e4-1079-11e7-847c-39dc4e52bf37.PNG)
![occurrenceidfound](https://cloud.githubusercontent.com/assets/7586150/24302668/00e97e5c-1079-11e7-9c43-12949a6915fc.PNG)

I could place a filter on the extended measurements or facts file as well as the occurrence file but in my opinion, valuable information is being lost to the user this way.

**Comment by dschigel** [comment]
Thank you @albenson-usgs for a concise summary of the issue and the OBIS perspectove. As you can see from the issue tags and as discussed at TDWG 2016, the problem is high on GBIF to-do list, and perhaps we can remove needs-clarification tag and find assignee for this. @ahahn-gbif @timrobertson100. This issue is a potential blocker for the sampling event mobilization campaign.

**Comment by dschigel** [comment]
Looking at the mobilization priority 2 https://www.gbif.org/news/4fkukOiVWoEMiKSa2meQ08/prioritizing-content-mobilization-in-the-gbif-network, the blocking nature of the absence issue needs some attention soon. Related - the OBIS paper on dealing with MeasurementOrFact and the Humboldt core paper are both out since we left this thread.

**Comment by Tasilee** [comment]
Hi Dmitry

Thanks for the link to the mobilization priorities. 'Gaps' have been a key
interest to me for some years.

Regards Priority 2, how do you see absences as a blocking issue (in GBIF)?
We handle them ok for analysis, but probably not so well more generally.
For example, if a taxon is mapped that contains both presences and
absences, you won't get colour differentiation unless you are in the
Spatial Portal and use faceting. We need some form of warning.

Lee









[image: untitled]



*Lee Belbin*

Science Advisor
Atlas of Living Australia
National Research Collections Australia, National Collections & Marine
Infrastructure, CSIRO *|* Clunies Ross Street, Acton ACT 2601 *|*
GPO Box 1700, Canberra  ACT  2601

Phone: +61 0419 374 133 *| *lee.belbin@csiro.au <leebelbin@gmail.com> *|*
www.ala.org.au *|*  http://www.csiro.au/en/Research/Collections

**Comment by dschigel** [comment]
Hi Lee, all,
Yes, I think absences issue is a blocker for sampling event mobilization, for the data publishers. Sampling event data are observation attempts, both successful (presence or abundance) and failed (absences) made within a 4-dimensional space: time, space, methods, and taxonomic target. It is a burden for a publisher, and a loss of value for the statistical analyses for the users, to ask removing the zeros in the publication process. What's worse, the current process actually reduces the original quality of the dataset, as it makes zeros (attempted, but failed observations) equal to empty cells (no data), not good.

**Comment by Tasilee** [comment]
Thanks Dmitry

We certainly need to handle 'absence' data in a systematic and transparent
way, and with the GBIF-ALA-iDigBio alignments, we need an agreed strategy.
My perspective is that we don't want to lose any data. Absences can be
valuable in many situations (e.g., modeling), but we need to make them
obvious. I'll discuss the issue here at the ALA.

Lee











[image: untitled]



*Lee Belbin*

Science Advisor
Atlas of Living Australia
National Research Collections Australia, National Collections & Marine
Infrastructure, CSIRO *|* Clunies Ross Street, Acton ACT 2601 *|*
GPO Box 1700, Canberra  ACT  2601

Phone: +61 0419 374 133 *| *lee.belbin@csiro.au <leebelbin@gmail.com> *|*
www.ala.org.au *|*  http://www.csiro.au/en/Research/Collections

**Comment by dhobern** [comment]
We need a whole new approach to incorporating sampling-event data (of which absences are a subset) into our mapping, API and data download structures. I can think of lots of visualisations that might be useful and that would show both relative abundance measures and absences.  We just need to make some progress.  Structuring the locality-event-taxon cube is the first step.

**Comment by dhobern** [comment]
@dschigel - I don't really believe that presentation of absences is a specially hard problem for us - provided we can avoid confusion around 0 or null being supplied as the abundance measure.  Unspecified values easily become absences.  In the context of sampling-event data, the key thing is the construction of the corresponding locality-event-taxon cube.

**Comment by albenson-usgs** [comment]
Happy to hear that this won't be too difficult to tackle. I have already published several datasets with absences so you can work against those if you need examples:
https://www.gbif.org/dataset/7606f22a-2121-4853-b3ab-2e693c6790dd
https://www.gbif.org/dataset/2501caa2-803b-4f6b-9e36-f9f5c72c0666
https://www.gbif.org/dataset/40bfac0e-a1b5-4c9e-862e-64327195f210
https://www.gbif.org/dataset/c684b43a-0147-459b-ae97-0e3d9757fb92

**Comment by albenson-usgs** [comment]
I wanted to add to this thread that @dschigel and I spoke at the TDWG 2017 annual meeting about this topic. He informed me that registering absence data in GBIF should not be an issue and that it's just a matter of how to display it appropriately. It sounds like there may be an sampling event display in the works. It also sounds like GBIF is interested in knowing what outputs users of sampling event data might want. I don't have a good answer right now but I'll ask around.

**Comment by albenson-usgs** [comment]
Just wondering what, if anything, has happened with this since 2017.

**Comment by Tasilee** [comment]
@albenson-usgs: A few weeks ago I listed all the records in the ALA that had something in dwc:samplingProtocol and retrieved 390,853 records, 99% of which were from 'Garden bird surveys'. Considering that around 30% of the 86 million ALA records are likely from systematic bird surveys, we have a huge loss of data. The only advances at the ALA end are a) now all record terms are at least indexed and exposed and b) I've prodded our Data Manager to pursue the full bird survey data. Until we have systematic surveys that are properly documented in Darwin Core, generating implied (but true) absences is impossible. The ALA does currently have 1,234,578 records from wwc:occurrenceStatus="absent". How any of this relates to current work to align GBIF and the ALA (and Living Atlases, iDigBio etc), one lives in hope.
